### DISCUSSION TITLE
Monaco in Mobile

### DISCUSSION THREAD

**Discussion by hemantasapkota** [microsoft/monaco-editor#3813] [Show and tell]
Hi team,

My name is Hemanta, and for the past year I've been working on a project that integrates monaco in react native. The integration was a huge success. 

Apps: 
https://itunes.apple.com/us/app/code-playground/id1452106609
~~https://itunes.apple.com/us/app/go-playground/id1437518275~~

I had to make several fixes to the editor to make it work decently inside react native, mostly around scroll view and mouse column computation. I've will submit a PR soon.

How can I attribute the use of the editor in my apps ?

Thanks

**Comment by donaldpipowitch** [comment]
This sounds really interesting. I'd love to see this. Do you still plan to submit a PR?

**Comment by k644606347** [comment]
This sounds great, I hope to use monaco on the mobile side.

**Comment by jdmunro** [comment]
@hemantasapkota I'd be very interested to see any changes you've made. Also looking for React Native integration. Do you have a fork of the repo somewhere?

**Comment by karlhorky** [comment]
This would be interesting!

Especially in case it would also handle the ability to long-press -> paste in mobile browsers like Android Chrome!

**Comment by Steve-Mcl** [comment]
@hemantasapkota what is the chances you could share the changes you made to make this mobile friendly?

**Comment by Jackman3005** [comment]
@hemantasapkota Our team is also interested in allowing users to edit code in a RN app. It's been a bit since you originally posted, is the solution you have still working well for you? Anything you're able to share to get others able to use Monaco w/ RN would be awesome!

**Comment by Jackman3005** [comment]
Someone on our team knew @hemantasapkota IRL and reached out (small world!). The feedback we got was that Monaco has issues on mobile that make it a poor choice right now. [This blog](https://blog.replit.com/codemirror) goes into details on that and suggests using [CodeMirror](https://codemirror.net/) which we've now implemented for ourselves and things are working okay for the moment, still not using many of the advanced features yet though.

Not trying to stop people from using Monaco (sorry maintainers!), just want to help people move forward with their goals, hopefully this helps.

**Comment by Lonli-Lokli** [comment]
Does the fact that issue got converted into discussion means that there are no plans for supporting mobile?
